File: src/VisualStudio/Tools/Test/Baselines/LTS_Northwind.cs

namespace BizLogic.Test
{
    using System;
    using System.Collections.Generic;
    using System.ComponentModel;
    using System.ComponentModel.DataAnnotations;
    using System.Data.Linq;
    using System.Linq;
    using DataTests.Northwind.LTS;
    using OpenRiaServices.Hosting;
    using OpenRiaServices.LinqToSql;
    using OpenRiaServices.Server;
    
    
    // Implements application logic using the NorthwindDataContext context.
    // TODO: Add your application logic to these methods or in additional methods.
    // TODO: Wire up authentication (Windows/ASP.NET Forms) and uncomment the following to disable anonymous access
    // Also consider adding roles to restrict access as appropriate.
    // [RequiresAuthentication]
    [EnableClientAccess()]
    public class LTS_Northwind : LinqToSqlDomainService<NorthwindDataContext>
    {
        
        // TODO:
        // Consider constraining the results of your query method.  If you need additional input you can
        // add parameters to this method or create additional query methods with different names.
        public IQueryable<Category> GetCategories()
        {
            return this.DataContext.Categories;
        }
        
        public void InsertCategory(Category category)
        {
            this.DataContext.Categories.InsertOnSubmit(category);
        }
        
        public void UpdateCategory(Category currentCategory)
        {
            this.DataContext.Categories.Attach(currentCategory, this.ChangeSet.GetOriginal(currentCategory));
        }
        
        public void DeleteCategory(Category category)
        {
            this.DataContext.Categories.Attach(category);
            this.DataContext.Categories.DeleteOnSubmit(category);
        }
        
        // TODO:
        // Consider constraining the results of your query method.  If you need additional input you can
        // add parameters to this method or create additional query methods with different names.
        public IQueryable<Customer> GetCustomers()
        {
            return this.DataContext.Customers;
        }
        
        public void InsertCustomer(Customer customer)
        {
            this.DataContext.Customers.InsertOnSubmit(customer);
        }
        
        public void UpdateCustomer(Customer currentCustomer)
        {
            this.DataContext.Customers.Attach(currentCustomer, this.ChangeSet.GetOriginal(currentCustomer));
        }
        
        public void DeleteCustomer(Customer customer)
        {
            this.DataContext.Customers.Attach(customer);
            this.DataContext.Customers.DeleteOnSubmit(customer);
        }
        
        // TODO:
        // Consider constraining the results of your query method.  If you need additional input you can
        // add parameters to this method or create additional query methods with different names.
        public IQueryable<CustomerCustomerDemo> GetCustomerCustomerDemos()
        {
            return this.DataContext.CustomerCustomerDemos;
        }
        
        public void InsertCustomerCustomerDemo(CustomerCustomerDemo customerCustomerDemo)
        {
            this.DataContext.CustomerCustomerDemos.InsertOnSubmit(customerCustomerDemo);
        }
        
        public void UpdateCustomerCustomerDemo(CustomerCustomerDemo currentCustomerCustomerDemo)
        {
            this.DataContext.CustomerCustomerDemos.Attach(currentCustomerCustomerDemo, this.ChangeSet.GetOriginal(currentCustomerCustomerDemo));
        }
        
        public void DeleteCustomerCustomerDemo(CustomerCustomerDemo customerCustomerDemo)
        {
            this.DataContext.CustomerCustomerDemos.Attach(customerCustomerDemo);
            this.DataContext.CustomerCustomerDemos.DeleteOnSubmit(customerCustomerDemo);
        }
        
        // TODO:
        // Consider constraining the results of your query method.  If you need additional input you can
        // add parameters to this method or create additional query methods with different names.
        public IQueryable<CustomerDemographic> GetCustomerDemographics()
        {
            return this.DataContext.CustomerDemographics;
        }
        
        public void InsertCustomerDemographic(CustomerDemographic customerDemographic)
        {
            this.DataContext.CustomerDemographics.InsertOnSubmit(customerDemographic);
        }
        
        public void UpdateCustomerDemographic(CustomerDemographic currentCustomerDemographic)
        {
            this.DataContext.CustomerDemographics.Attach(currentCustomerDemographic, this.ChangeSet.GetOriginal(currentCustomerDemographic));
        }
        
        public void DeleteCustomerDemographic(CustomerDemographic customerDemographic)
        {
            this.DataContext.CustomerDemographics.Attach(customerDemographic);
            this.DataContext.CustomerDemographics.DeleteOnSubmit(customerDemographic);
        }
        
        // TODO:
        // Consider constraining the results of your query method.  If you need additional input you can
        // add parameters to this method or create additional query methods with different names.
        public IQueryable<Employee> GetEmployees()
        {
            return this.DataContext.Employees;
        }
        
        public void InsertEmployee(Employee employee)
        {
            this.DataContext.Employees.InsertOnSubmit(employee);
        }
        
        public void UpdateEmployee(Employee currentEmployee)
        {
            this.DataContext.Employees.Attach(currentEmployee, this.ChangeSet.GetOriginal(currentEmployee));
        }
        
        public void DeleteEmployee(Employee employee)
        {
            this.DataContext.Employees.Attach(employee);
            this.DataContext.Employees.DeleteOnSubmit(employee);
        }
        
        // TODO:
        // Consider constraining the results of your query method.  If you need additional input you can
        // add parameters to this method or create additional query methods with different names.
        public IQueryable<EmployeeTerritory> GetEmployeeTerritories()
        {
            return this.DataContext.EmployeeTerritories;
        }
        
        public void InsertEmployeeTerritory(EmployeeTerritory employeeTerritory)
        {
            this.DataContext.EmployeeTerritories.InsertOnSubmit(employeeTerritory);
        }
        
        public void UpdateEmployeeTerritory(EmployeeTerritory currentEmployeeTerritory)
        {
            this.DataContext.EmployeeTerritories.Attach(currentEmployeeTerritory, this.ChangeSet.GetOriginal(currentEmployeeTerritory));
        }
        
        public void DeleteEmployeeTerritory(EmployeeTerritory employeeTerritory)
        {
            this.DataContext.EmployeeTerritories.Attach(employeeTerritory);
            this.DataContext.EmployeeTerritories.DeleteOnSubmit(employeeTerritory);
        }
        
        // TODO:
        // Consider constraining the results of your query method.  If you need additional input you can
        // add parameters to this method or create additional query methods with different names.
        public IQueryable<Order> GetOrders()
        {
            return this.DataContext.Orders;
        }
        
        public void InsertOrder(Order order)
        {
            this.DataContext.Orders.InsertOnSubmit(order);
        }
        
        public void UpdateOrder(Order currentOrder)
        {
            this.DataContext.Orders.Attach(currentOrder, this.ChangeSet.GetOriginal(currentOrder));
        }
        
        public void DeleteOrder(Order order)
        {
            this.DataContext.Orders.Attach(order);
            this.DataContext.Orders.DeleteOnSubmit(order);
        }
        
        // TODO:
        // Consider constraining the results of your query method.  If you need additional input you can
        // add parameters to this method or create additional query methods with different names.
        public IQueryable<Order_Detail> GetOrder_Details()
        {
            return this.DataContext.Order_Details;
        }
        
        public void InsertOrder_Detail(Order_Detail order_Detail)
        {
            this.DataContext.Order_Details.InsertOnSubmit(order_Detail);
        }
        
        public void UpdateOrder_Detail(Order_Detail currentOrder_Detail)
        {
            this.DataContext.Order_Details.Attach(currentOrder_Detail, this.ChangeSet.GetOriginal(currentOrder_Detail));
        }
        
        public void DeleteOrder_Detail(Order_Detail order_Detail)
        {
            this.DataContext.Order_Details.Attach(order_Detail);
            this.DataContext.Order_Details.DeleteOnSubmit(order_Detail);
        }
        
        // TODO:
        // Consider constraining the results of your query method.  If you need additional input you can
        // add parameters to this method or create additional query methods with different names.
        public IQueryable<Product> GetProducts()
        {
            return this.DataContext.Products;
        }
        
        public void InsertProduct(Product product)
        {
            this.DataContext.Products.InsertOnSubmit(product);
        }
        
        public void UpdateProduct(Product currentProduct)
        {
            this.DataContext.Products.Attach(currentProduct, this.ChangeSet.GetOriginal(currentProduct));
        }
        
        public void DeleteProduct(Product product)
        {
            this.DataContext.Products.Attach(product);
            this.DataContext.Products.DeleteOnSubmit(product);
        }
        
        // TODO:
        // Consider constraining the results of your query method.  If you need additional input you can
        // add parameters to this method or create additional query methods with different names.
        public IQueryable<Region> GetRegions()
        {
            return this.DataContext.Regions;
        }
        
        public void InsertRegion(Region region)
        {
            this.DataContext.Regions.InsertOnSubmit(region);
        }
        
        public void UpdateRegion(Region currentRegion)
        {
            this.DataContext.Regions.Attach(currentRegion, this.ChangeSet.GetOriginal(currentRegion));
        }
        
        public void DeleteRegion(Region region)
        {
            this.DataContext.Regions.Attach(region);
            this.DataContext.Regions.DeleteOnSubmit(region);
        }
        
        // TODO:
        // Consider constraining the results of your query method.  If you need additional input you can
        // add parameters to this method or create additional query methods with different names.
        public IQueryable<Shipper> GetShippers()
        {
            return this.DataContext.Shippers;
        }
        
        public void InsertShipper(Shipper shipper)
        {
            this.DataContext.Shippers.InsertOnSubmit(shipper);
        }
        
        public void UpdateShipper(Shipper currentShipper)
        {
            this.DataContext.Shippers.Attach(currentShipper, this.ChangeSet.GetOriginal(currentShipper));
        }
        
        public void DeleteShipper(Shipper shipper)
        {
            this.DataContext.Shippers.Attach(shipper);
            this.DataContext.Shippers.DeleteOnSubmit(shipper);
        }
        
        // TODO:
        // Consider constraining the results of your query method.  If you need additional input you can
        // add parameters to this method or create additional query methods with different names.
        public IQueryable<Supplier> GetSuppliers()
        {
            return this.DataContext.Suppliers;
        }
        
        public void InsertSupplier(Supplier supplier)
        {
            this.DataContext.Suppliers.InsertOnSubmit(supplier);
        }
        
        public void UpdateSupplier(Supplier currentSupplier)
        {
            this.DataContext.Suppliers.Attach(currentSupplier, this.ChangeSet.GetOriginal(currentSupplier));
        }
        
        public void DeleteSupplier(Supplier supplier)
        {
            this.DataContext.Suppliers.Attach(supplier);
            this.DataContext.Suppliers.DeleteOnSubmit(supplier);
        }
        
        // TODO:
        // Consider constraining the results of your query method.  If you need additional input you can
        // add parameters to this method or create additional query methods with different names.
        public IQueryable<Territory> GetTerritories()
        {
            return this.DataContext.Territories;
        }
        
        public void InsertTerritory(Territory territory)
        {
            this.DataContext.Territories.InsertOnSubmit(territory);
        }
        
        public void UpdateTerritory(Territory currentTerritory)
        {
            this.DataContext.Territories.Attach(currentTerritory, this.ChangeSet.GetOriginal(currentTerritory));
        }
        
        public void DeleteTerritory(Territory territory)
        {
            this.DataContext.Territories.Attach(territory);
            this.DataContext.Territories.DeleteOnSubmit(territory);
        }
    }
}

File: src/VisualStudio/Tools/Test/Baselines/LTS_Northwind.vb

Option Compare Binary
Option Infer On
Option Strict On
Option Explicit On

Imports DataTests.Northwind.LTS
Imports OpenRiaServices.LinqToSql
Imports OpenRiaServices.Server
Imports System
Imports System.Collections.Generic
Imports System.ComponentModel
Imports System.ComponentModel.DataAnnotations
Imports System.Data.Linq
Imports System.Linq

Namespace BizLogic.Test
    
    'Implements application logic using the NorthwindDataContext context.
    ' TODO: Add your application logic to these methods or in additional methods.
    ' TODO: Wire up authentication (Windows/ASP.NET Forms) and uncomment the following to disable anonymous access
    ' Also consider adding roles to restrict access as appropriate.
    '<RequiresAuthentication> _
    <EnableClientAccess()>  _
    Public Class LTS_Northwind
        Inherits LinqToSqlDomainService(Of NorthwindDataContext)
        
        'TODO:
        ' Consider constraining the results of your query method.  If you need additional input you can
        ' add parameters to this method or create additional query methods with different names.
        Public Function GetCategories() As IQueryable(Of Category)
            Return Me.DataContext.Categories
        End Function
        
        Public Sub InsertCategory(ByVal category As Category)
            Me.DataContext.Categories.InsertOnSubmit(category)
        End Sub
        
        Public Sub UpdateCategory(ByVal currentCategory As Category)
            Me.DataContext.Categories.Attach(currentCategory, Me.ChangeSet.GetOriginal(currentCategory))
        End Sub
        
        Public Sub DeleteCategory(ByVal category As Category)
            Me.DataContext.Categories.Attach(category)
            Me.DataContext.Categories.DeleteOnSubmit(category)
        End Sub
        
        'TODO:
        ' Consider constraining the results of your query method.  If you need additional input you can
        ' add parameters to this method or create additional query methods with different names.
        Public Function GetCustomers() As IQueryable(Of Customer)
            Return Me.DataContext.Customers
        End Function
        
        Public Sub InsertCustomer(ByVal customer As Customer)
            Me.DataContext.Customers.InsertOnSubmit(customer)
        End Sub
        
        Public Sub UpdateCustomer(ByVal currentCustomer As Customer)
            Me.DataContext.Customers.Attach(currentCustomer, Me.ChangeSet.GetOriginal(currentCustomer))
        End Sub
        
        Public Sub DeleteCustomer(ByVal customer As Customer)
            Me.DataContext.Customers.Attach(customer)
            Me.DataContext.Customers.DeleteOnSubmit(customer)
        End Sub
        
        'TODO:
        ' Consider constraining the results of your query method.  If you need additional input you can
        ' add parameters to this method or create additional query methods with different names.
        Public Function GetCustomerCustomerDemos() As IQueryable(Of CustomerCustomerDemo)
            Return Me.DataContext.CustomerCustomerDemos
        End Function
        
        Public Sub InsertCustomerCustomerDemo(ByVal customerCustomerDemo As CustomerCustomerDemo)
            Me.DataContext.CustomerCustomerDemos.InsertOnSubmit(customerCustomerDemo)
        End Sub
        
        Public Sub UpdateCustomerCustomerDemo(ByVal currentCustomerCustomerDemo As CustomerCustomerDemo)
            Me.DataContext.CustomerCustomerDemos.Attach(currentCustomerCustomerDemo, Me.ChangeSet.GetOriginal(currentCustomerCustomerDemo))
        End Sub
        
        Public Sub DeleteCustomerCustomerDemo(ByVal customerCustomerDemo As CustomerCustomerDemo)
            Me.DataContext.CustomerCustomerDemos.Attach(customerCustomerDemo)
            Me.DataContext.CustomerCustomerDemos.DeleteOnSubmit(customerCustomerDemo)
        End Sub
        
        'TODO:
        ' Consider constraining the results of your query method.  If you need additional input you can
        ' add parameters to this method or create additional query methods with different names.
        Public Function GetCustomerDemographics() As IQueryable(Of CustomerDemographic)
            Return Me.DataContext.CustomerDemographics
        End Function
        
        Public Sub InsertCustomerDemographic(ByVal customerDemographic As CustomerDemographic)
            Me.DataContext.CustomerDemographics.InsertOnSubmit(customerDemographic)
        End Sub
        
        Public Sub UpdateCustomerDemographic(ByVal currentCustomerDemographic As CustomerDemographic)
            Me.DataContext.CustomerDemographics.Attach(currentCustomerDemographic, Me.ChangeSet.GetOriginal(currentCustomerDemographic))
        End Sub
        
        Public Sub DeleteCustomerDemographic(ByVal customerDemographic As CustomerDemographic)
            Me.DataContext.CustomerDemographics.Attach(customerDemographic)
            Me.DataContext.CustomerDemographics.DeleteOnSubmit(customerDemographic)
        End Sub
        
        'TODO:
        ' Consider constraining the results of your query method.  If you need additional input you can
        ' add parameters to this method or create additional query methods with different names.
        Public Function GetEmployees() As IQueryable(Of Employee)
            Return Me.DataContext.Employees
        End Function
        
        Public Sub InsertEmployee(ByVal employee As Employee)
            Me.DataContext.Employees.InsertOnSubmit(employee)
        End Sub
        
        Public Sub UpdateEmployee(ByVal currentEmployee As Employee)
            Me.DataContext.Employees.Attach(currentEmployee, Me.ChangeSet.GetOriginal(currentEmployee))
        End Sub
        
        Public Sub DeleteEmployee(ByVal employee As Employee)
            Me.DataContext.Employees.Attach(employee)
            Me.DataContext.Employees.DeleteOnSubmit(employee)
        End Sub
        
        'TODO:
        ' Consider constraining the results of your query method.  If you need additional input you can
        ' add parameters to this method or create additional query methods with different names.
        Public Function GetEmployeeTerritories() As IQueryable(Of EmployeeTerritory)
            Return Me.DataContext.EmployeeTerritories
        End Function
        
        Public Sub InsertEmployeeTerritory(ByVal employeeTerritory As EmployeeTerritory)
            Me.DataContext.EmployeeTerritories.InsertOnSubmit(employeeTerritory)
        End Sub
        
        Public Sub UpdateEmployeeTerritory(ByVal currentEmployeeTerritory As EmployeeTerritory)
            Me.DataContext.EmployeeTerritories.Attach(currentEmployeeTerritory, Me.ChangeSet.GetOriginal(currentEmployeeTerritory))
        End Sub
        
        Public Sub DeleteEmployeeTerritory(ByVal employeeTerritory As EmployeeTerritory)
            Me.DataContext.EmployeeTerritories.Attach(employeeTerritory)
            Me.DataContext.EmployeeTerritories.DeleteOnSubmit(employeeTerritory)
        End Sub
        
        'TODO:
        ' Consider constraining the results of your query method.  If you need additional input you can
        ' add parameters to this method or create additional query methods with different names.
        Public Function GetOrders() As IQueryable(Of Order)
            Return Me.DataContext.Orders
        End Function
        
        Public Sub InsertOrder(ByVal order As Order)
            Me.DataContext.Orders.InsertOnSubmit(order)
        End Sub
        
        Public Sub UpdateOrder(ByVal currentOrder As Order)
            Me.DataContext.Orders.Attach(currentOrder, Me.ChangeSet.GetOriginal(currentOrder))
        End Sub
        
        Public Sub DeleteOrder(ByVal order As Order)
            Me.DataContext.Orders.Attach(order)
            Me.DataContext.Orders.DeleteOnSubmit(order)
        End Sub
        
        'TODO:
        ' Consider constraining the results of your query method.  If you need additional input you can
        ' add parameters to this method or create additional query methods with different names.
        Public Function GetOrder_Details() As IQueryable(Of Order_Detail)
            Return Me.DataContext.Order_Details
        End Function
        
        Public Sub InsertOrder_Detail(ByVal order_Detail As Order_Detail)
            Me.DataContext.Order_Details.InsertOnSubmit(order_Detail)
        End Sub
        
        Public Sub UpdateOrder_Detail(ByVal currentOrder_Detail As Order_Detail)
            Me.DataContext.Order_Details.Attach(currentOrder_Detail, Me.ChangeSet.GetOriginal(currentOrder_Detail))
        End Sub
        
        Public Sub DeleteOrder_Detail(ByVal order_Detail As Order_Detail)
            Me.DataContext.Order_Details.Attach(order_Detail)
            Me.DataContext.Order_Details.DeleteOnSubmit(order_Detail)
        End Sub
        
        'TODO:
        ' Consider constraining the results of your query method.  If you need additional input you can
        ' add parameters to this method or create additional query methods with different names.
        Public Function GetProducts() As IQueryable(Of Product)
            Return Me.DataContext.Products
        End Function
        
        Public Sub InsertProduct(ByVal product As Product)
            Me.DataContext.Products.InsertOnSubmit(product)
        End Sub
        
        Public Sub UpdateProduct(ByVal currentProduct As Product)
            Me.DataContext.Products.Attach(currentProduct, Me.ChangeSet.GetOriginal(currentProduct))
        End Sub
        
        Public Sub DeleteProduct(ByVal product As Product)
            Me.DataContext.Products.Attach(product)
            Me.DataContext.Products.DeleteOnSubmit(product)
        End Sub
        
        'TODO:
        ' Consider constraining the results of your query method.  If you need additional input you can
        ' add parameters to this method or create additional query methods with different names.
        Public Function GetRegions() As IQueryable(Of Region)
            Return Me.DataContext.Regions
        End Function
        
        Public Sub InsertRegion(ByVal region As Region)
            Me.DataContext.Regions.InsertOnSubmit(region)
        End Sub
        
        Public Sub UpdateRegion(ByVal currentRegion As Region)
            Me.DataContext.Regions.Attach(currentRegion, Me.ChangeSet.GetOriginal(currentRegion))
        End Sub
        
        Public Sub DeleteRegion(ByVal region As Region)
            Me.DataContext.Regions.Attach(region)
            Me.DataContext.Regions.DeleteOnSubmit(region)
        End Sub
        
        'TODO:
        ' Consider constraining the results of your query method.  If you need additional input you can
        ' add parameters to this method or create additional query methods with different names.
        Public Function GetShippers() As IQueryable(Of Shipper)
            Return Me.DataContext.Shippers
        End Function
        
        Public Sub InsertShipper(ByVal shipper As Shipper)
            Me.DataContext.Shippers.InsertOnSubmit(shipper)
        End Sub
        
        Public Sub UpdateShipper(ByVal currentShipper As Shipper)
            Me.DataContext.Shippers.Attach(currentShipper, Me.ChangeSet.GetOriginal(currentShipper))
        End Sub
        
        Public Sub DeleteShipper(ByVal shipper As Shipper)
            Me.DataContext.Shippers.Attach(shipper)
            Me.DataContext.Shippers.DeleteOnSubmit(shipper)
        End Sub
        
        'TODO:
        ' Consider constraining the results of your query method.  If you need additional input you can
        ' add parameters to this method or create additional query methods with different names.
        Public Function GetSuppliers() As IQueryable(Of Supplier)
            Return Me.DataContext.Suppliers
        End Function
        
        Public Sub InsertSupplier(ByVal supplier As Supplier)
            Me.DataContext.Suppliers.InsertOnSubmit(supplier)
        End Sub
        
        Public Sub UpdateSupplier(ByVal currentSupplier As Supplier)
            Me.DataContext.Suppliers.Attach(currentSupplier, Me.ChangeSet.GetOriginal(currentSupplier))
        End Sub
        
        Public Sub DeleteSupplier(ByVal supplier As Supplier)
            Me.DataContext.Suppliers.Attach(supplier)
            Me.DataContext.Suppliers.DeleteOnSubmit(supplier)
        End Sub
        
        'TODO:
        ' Consider constraining the results of your query method.  If you need additional input you can
        ' add parameters to this method or create additional query methods with different names.
        Public Function GetTerritories() As IQueryable(Of Territory)
            Return Me.DataContext.Territories
        End Function
        
        Public Sub InsertTerritory(ByVal territory As Territory)
            Me.DataContext.Territories.InsertOnSubmit(territory)
        End Sub
        
        Public Sub UpdateTerritory(ByVal currentTerritory As Territory)
            Me.DataContext.Territories.Attach(currentTerritory, Me.ChangeSet.GetOriginal(currentTerritory))
        End Sub
        
        Public Sub DeleteTerritory(ByVal territory As Territory)
            Me.DataContext.Territories.Attach(territory)
            Me.DataContext.Territories.DeleteOnSubmit(territory)
        End Sub
    End Class
End Namespace
